client/internal/peer/conn.go

```diff
@@ -343,6 +387,22 @@ func (conn *Conn) cleanup() error {
 
 	conn.status = StatusDisconnected
 
+	peerState, err := conn.statusRecorder.GetPeerStatus(conn.config.Key)
+	if err == nil {
+		peerState.ConnStatus = conn.status.String()
+		peerState.ConnStatusUpdate = time.Now()
+		peerState.Direct = false
+		peerState.Relayed = false
+		peerState.RemoteIceCandidateType = ""
+		peerState.LocalIceCandidateType = ""
+		err = conn.statusRecorder.UpdatePeerStatus(peerState)
```

Comment:
2. I would rather not query the status and create a separate method in the statusRecorder to create a status Disconnected and not pollute the Conn